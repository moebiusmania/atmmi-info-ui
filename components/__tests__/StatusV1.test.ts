import { mount } from "@vue/test-utils";
import { expect, test, vi, describe } from "vitest";
import StatusV1 from "../StatusV1.vue";

// Mock the Card component
vi.mock("../Card.vue", () => ({
  default: {
    name: "Card",
    template: '<div class="card-mock"><slot /></div>',
    props: ["title"],
  },
}));

// Mock useFetch
const mockUseFetch = vi.fn();
vi.mock("#imports", () => ({
  useFetch: () => mockUseFetch(),
}));

describe("<StatusV1 />", () => {
  test("renders loading state correctly", async () => {
    mockUseFetch.mockReturnValue({
      status: { value: "pending" },
      data: { value: [] },
      error: { value: null },
    });

    const wrapper = mount(StatusV1);
    expect(wrapper.html()).toMatchSnapshot("loading");
  });

  test("renders data correctly", async () => {
    mockUseFetch.mockReturnValue({
      status: { value: "success" },
      data: {
        value: [
          { line: "M1", text: "Linea 1", status: "regolare" },
          { line: "M2", text: "Linea 2", status: "tratta sospesa" },
        ],
      },
      error: { value: null },
    });

    const wrapper = mount(StatusV1);
    expect(wrapper.html()).toMatchSnapshot("with-data");
  });
});
