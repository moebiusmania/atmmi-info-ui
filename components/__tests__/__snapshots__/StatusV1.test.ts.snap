// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<StatusV1 /> > renders data correctly > with-data 1`] = `""`;

exports[`<StatusV1 /> > renders loading state correctly > loading 1`] = `""`;
